classes/PaymentModule.php

```diff
@@ -1250,4 +1238,36 @@ protected function createOrderCartRules(
 
         return $cart_rules_list;
     }
+
+    /**
+     * @param $order
+     * @param Message $update_message
+     */
+    private function copyMessageIntoCustomerMessage(Order $order, Message $update_message): void
+    {
+        if (null === $this->customerThread) {
```

Comment:
I don't see how it can be not null. Unless we call validateOrder twice in the same process

Reply:
The function `copyMessageIntoCustomerMessage()` is called twice in validateOrder().

The 1st time is when `validateOrder()` was called with a not-null `$message` parameter (this is the bug this PR targets).
The 2nd time is when the customer submitted a message with his order.

Sometimes only the 1st call will be done, sometimes, only the 2nd. Sometimes none, sometimes both.

This `if (null === $this->customerThread)` allows us to avoid creating 2 Customer Threads when we are in the "both" usecase. The 2 messages are then added into the same thread.